src/context/index.tsx

```diff
@@ -0,0 +1,36 @@
+import React, {
+  useState, useEffect, useMemo,
+} from 'react';
+import { enqueueSnackbar } from 'notistack';
+import { AxiosError } from 'axios';
+import { axiosInstance } from '../utils/apis';
+import { AppContextProps, UserData } from './types';
+import userDataContext from './contextData';
+
+const AppContext: React.FC<AppContextProps> = ({ children }) => {
+  const [userData, setUserData] = useState<UserData[]>([]);
+  useEffect(() => {
+    const getUserData = async () => {
+      try {
+        const response = await axiosInstance.get('auth/');
+        setUserData(response.data);
+      } catch (error) {
+        const axiosError = error as AxiosError;
+        enqueueSnackbar(axiosError.message, { variant: 'error' });
+      }
+    };
+    getUserData();
+  }, []);
+  const contextValue = useMemo(() => ({ userData, setUserData }), [userData, setUserData]);
+

```

Comment:
can we create a loading state and if it's loading return loading instead of userDataContext?